ar_track_alvar/nodes/IndividualMarkers.cpp

```diff
@@ -194,7 +194,32 @@ int PlaneFitPoseImprovement(int id, const ARCloud &corners_3D, ARCloud::Ptr sele
   gm::PoseStamped pose;
   pose.header.stamp = pcl_conversions::fromPCL(cloud.header).stamp;
   pose.header.frame_id = cloud.header.frame_id;
-  pose.pose.position = ata::centroid(*res.inliers);
+
+  // Project 3D corners to plane
+  if (res.coeffs.values.size() < 4) {
+    ROS_DEBUG_STREAM("Invalid number of plane coefficients.");
+    return -1;
+  }
+
+  ARCloud::Ptr corners_3D_ptr(new ARCloud());
+  *corners_3D_ptr = corners_3D;
+  pcl::ModelCoefficients::Ptr coeffs_ptr(new pcl::ModelCoefficients ());
+  *coeffs_ptr = res.coeffs;
+  ARCloud::Ptr projected_corners_3D = ata::projectCloudInPlane(corners_3D_ptr, coeffs_ptr);
+
+  // Compute center of projected corners
```

Comment:
You can still call `ata::centroid()`, but on the corners instead of on all the inliers.

Reply:
I looked into this and I don't think this could add more information. The only thing that is detected in the 2D image is the location of the four corners of the 'blob'. The other coordinates are merely used for testing the AR pattern and not detected in 2D but computed from the four other corners.